src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -556,6 +559,24 @@ def _get_hash_machine_id():
     return str(uuid.UUID(_get_hash_mac_address()[0:32]))
 
 
+@decorators.suppress_all_exceptions(fallback_return='')
+def _get_distro_name():
+    try:
+        import distro
+        return distro.name(pretty=True)
+    except ImportError:
+        return ''
+
+
+@decorators.suppress_all_exceptions(fallback_return='')
+def _get_distro_version():
+    try:
+        import distro
+        return distro.version(pretty=True)
+    except ImportError:
+        return ''
+
```

Comment:
Should we add `pretty=True`？
If it is added, name and version is `Ubuntu 22.04.1 LTS`, `22.04 (jammy)`.
Without it, they become `Ubuntu`, `22.04`.
I think the later one is easy to use if you want to get aggregate data.

Reply:
I added `pretty=True` to collect as much data as possible. But I'm also open for the short option according to how we really need to use the info.

@jiasli What do you think?

Reply:
Maybe we can keep `pretty=True` for `distro.name` but omit for `distro.version`?

Reply:
```
$ python
>>> import distro
>>> distro.name()
'Ubuntu'
>>> distro.name(pretty=True)
'Ubuntu 20.04.4 LTS'
>>> distro.version()
'20.04'
>>> distro.version(pretty=True)
'20.04 (focal)'
>>> distro.id()
'ubuntu'
```

The above information is retrieved from

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

Reply:
According to the source code os `distro`:

https://github.com/python-distro/distro/blob/6e41b8d5b0f3b962c7fcfff37271fcb4398c2558/src/distro/distro.py#L887

```py
        versions = [
            self.os_release_attr("version_id"),
```

`version_id` is the most preferred result for `distro.version()`.